crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -671,8 +675,19 @@ impl RoomEventCacheInner {
             self.pagination.token_notifier.notify_one();
         }
 
-        let _ =
-            self.sender.send(RoomEventCacheUpdate::Append { events, ephemeral, ambiguity_changes });
+        let _ = self.send_grouped_updates_for_events(events, ephemeral, ambiguity_changes);
+
+        Ok(())
+    }
+
+    fn send_grouped_updates_for_events(
+        &self,
+        events: Vec<SyncTimelineEvent>,
+        ephemeral: Vec<Raw<AnySyncEphemeralRoomEvent>>,
+        ambiguity_changes: BTreeMap<OwnedEventId, AmbiguityChange>,
+    ) -> StdResult<(), SendError<RoomEventCacheUpdate>> {
+        self.sender.send(RoomEventCacheUpdate::Append { events, ephemeral })?;
+        self.sender.send(RoomEventCacheUpdate::Members { ambiguity_changes })?;
```

Comment:
I'm not sure introducing a new function is worth it to do so little, especially as it's happening only once. But if you care strongly about having a new function, can you document that it exists to ensure the ordering invariant, please? (as a code comment above the first `send()` maybe?)

Reply:
Inlined the function body.